controllers/cloud_config_sync_controller_test.go

```diff
@@ -0,0 +1,275 @@
+
+	var infraCloudConfig *corev1.ConfigMap
+	var managedCloudConfig *corev1.ConfigMap
+	var syncedCloudConfigMap *corev1.ConfigMap
+
+	var reconciler *CloudConfigReconciler
+
+	syncedConfigMapKey := client.ObjectKey{Namespace: targetNamespaceName, Name: cloudConfigMapName}
+
+	BeforeEach(func() {
+		By("Setting up a new manager")
+		mgr, err := manager.New(cfg, manager.Options{MetricsBindAddress: "0"})
+		Expect(err).NotTo(HaveOccurred())
+
+		reconciler = &CloudConfigReconciler{
+			Client:          cl,
+			Scheme:          scheme.Scheme,
+			Recorder:        rec,
+			TargetNamespace: targetNamespaceName,
+		}
+		Expect(reconciler.SetupWithManager(mgr)).To(Succeed())
+
+		By("Creating Infra resource")
+		Expect(cl.Create(ctx, makeInfrastrucutreResource())).To(Succeed())
+
+		By("Creating needed ConfigMaps")
+		infraCloudConfig = makeInfraCloudConifg()
+		managedCloudConfig = makeManagedCloudConifg()
+		Expect(cl.Create(ctx, infraCloudConfig)).To(Succeed())
+		Expect(cl.Create(ctx, managedCloudConfig)).To(Succeed())
+
+		var mgrCtx context.Context
+		mgrCtx, mgrCtxCancel = context.WithCancel(ctx)
+		mgrStopped = make(chan struct{})
+
+		By("Starting the manager")
+		go func() {
+			defer GinkgoRecover()
+			defer close(mgrStopped)
+
+			Expect(mgr.Start(mgrCtx)).To(Succeed())
+		}()
+	})
+
+	AfterEach(func() {
+		By("Closing the manager")
+		mgrCtxCancel()
+		Eventually(mgrStopped, timeout).Should(BeClosed())
+
+		By("Cleanup resources")
+		deleteOptions := &client.DeleteOptions{
+			GracePeriodSeconds: pointer.Int64(0),
+		}
+
+		if infraCloudConfig != nil {
+			Expect(cl.Delete(ctx, infraCloudConfig, deleteOptions)).To(Succeed())
+			Eventually(
+				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infraCloudConfig), &corev1.ConfigMap{})),
+			).Should(BeTrue())
+		}
+
+		if managedCloudConfig != nil {
+			Expect(cl.Delete(ctx, managedCloudConfig, deleteOptions)).To(Succeed())
+			Eventually(
+				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(managedCloudConfig), &corev1.ConfigMap{})),
+			).Should(BeTrue())
+		}
+
+		if syncedCloudConfigMap != nil {
+			Expect(cl.Delete(ctx, syncedCloudConfigMap, deleteOptions)).To(Succeed())
+			Eventually(
+				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(syncedCloudConfigMap), &corev1.Namespace{})),
+			).Should(BeTrue())
+		}
+
+		infraCloudConfig = nil
+		managedCloudConfig = nil
+		syncedCloudConfigMap = nil
+
+		infra := makeInfrastrucutreResource()
+		Expect(cl.Delete(ctx, infra)).To(Succeed())
+		Eventually(
+			apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infra), infra)),
+		).Should(BeTrue())
+	})
+
+	It("config should be synced up after first reconcile", func() {
+		syncedCloudConfigMap := &corev1.ConfigMap{}
+		Eventually(func() error {
+			return cl.Get(ctx, syncedConfigMapKey, syncedCloudConfigMap)
+		}).Should(Succeed())
+		Expect(syncedCloudConfigMap.Data[defaultConfigKey]).Should(BeEquivalentTo("bar"))
```

Comment:
Perhaps move the equivalence check inside the eventually? Only needs one assertion here as we discussed yesterday